packages/datastore/src/request.js

```diff
@@ -1057,7 +1083,34 @@ DatastoreRequest.prototype.request_ = function(protoOpts, reqOpts, callback) {
     };
   }
 
-  this.request(protoOpts, reqOpts, callback);
+  datastore.auth.getProjectId(function(err, projectId) {
+    if (err) {
+      callback(err);
+      return;
+    }
+
+    var gaxClient = datastore.api[config.client];
+
+    var clientOptions = extend({
+      servicePath: datastore.baseUrl_,
+      port: datastore.port_
+    }, datastore.options);
+
+    if (datastore.customEndpoint_) {
+      clientOptions.sslCreds = grpc.credentials.createInsecure();
```

Comment:
Ah agreed. Got a little confused thinking the error was happening at that line specifically. I'll try and get that PR merged soon.